Meadowlark-js/packages/meadowlark-core/src/handler/NaturalKeyExtractor.ts

```diff
@@ -88,9 +88,10 @@ function naturalKeysFrom(
  * extracts the natural key information from the JSON body. Also extracts security information, if any.
  */
 export function extractNaturalKey(entity: TopLevelEntity, body: object): NaturalKeyWithSecurity {
-  const naturalKeysWithSecurity: NaturalKeyWithSecurity[] = (entity.data
-    .meadowlark as EntityMeadowlarkData).apiMapping.identityReferenceComponents.flatMap(
-    (identityReferenceComponent: ReferenceComponent) => naturalKeysFrom(identityReferenceComponent, body, entity, []),
+  const naturalKeysWithSecurity: NaturalKeyWithSecurity[] = (
+    entity.data.meadowlark as EntityMeadowlarkData
+  ).apiMapping.identityReferenceComponents.flatMap((identityReferenceComponent: ReferenceComponent) =>
+    naturalKeysFrom(identityReferenceComponent, body, entity, []),
```

Comment:
That's a lot of code to read in this one statement. Makes me wonder if some of this "work" should be done inside another class? I don't know the object model here anymore, so this question might not make any sense.

Reply:
This statement looks more complicated than it is. Everything before the `flatMap` is just referencing the `identityReferenceComponents` of the `entity`, and `flatMap` is calling a single function `naturalKeysFrom`. The type annotations make it look somewhat verbose.